app/src/main/java/org/mascota/ui/view/content/detail/adapter/SwipeHelperCallback.kt

```diff
@@ -0,0 +1,140 @@
+package org.mascota.ui.view.content.detail.adapter
+
+import android.graphics.Canvas
+import android.icu.lang.UCharacter.IndicPositionalCategory.LEFT
+import android.os.Build
+import android.view.View
+import androidx.annotation.RequiresApi
+import androidx.recyclerview.widget.ItemTouchHelper
+import androidx.recyclerview.widget.RecyclerView
+import java.lang.Float.max
+import java.lang.Float.min
+
+class SwipeHelperCallback : ItemTouchHelper.Callback() {
+
+    private var curPosition: Int? = null
+    private var prePosition: Int? = null
+    private var curDx = 0f
+    private var clamp = 0f
+
+    override fun getMovementFlags(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder
+    ): Int {
+        // Drag와 Swipe 방향을 결정 Drag는 사용하지 않아 0, Swipe의 경우는 LEFT 사용가능하도록 설정
+        return makeMovementFlags(0, LEFT)
+    }
+
+    override fun onMove(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        target: RecyclerView.ViewHolder
+    ) = false
+
+    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {}
+
+    override fun clearView(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder) {
+        curDx = 0f
+        prePosition = viewHolder.bindingAdapterPosition
+        getDefaultUIUtil().clearView(getView(viewHolder))
+    }
+
+    override fun onSelectedChanged(viewHolder: RecyclerView.ViewHolder?, actionState: Int) {
+        viewHolder?.let {
+            curPosition = viewHolder.bindingAdapterPosition
+            getDefaultUIUtil().onSelected(getView(it))
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onChildDraw(
+        c: Canvas,
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        dX: Float,
+        dY: Float,
+        actionState: Int,
+        isCurrentlyActive: Boolean
+    ) {
+        if (actionState == ItemTouchHelper.ACTION_STATE_SWIPE) {
+            val view = getView(viewHolder)
+            val isClamped = getTag(viewHolder)
+            val x = clampViewPositionHorizontal(view, dX, isClamped, isCurrentlyActive)
+
+            curDx = x
+
+            getDefaultUIUtil().onDraw(
+                c,
+                recyclerView,
+                view,
+                x,
+                dY,
+                actionState,
+                isCurrentlyActive
+            )
+        }
+    }
+
+    private fun getView(viewHolder: RecyclerView.ViewHolder): View {
+        return (viewHolder as ContentDetailDiaryAdapter.ContentDetailDiaryViewHolder).binding.clItemDiaryTop
+    }
```

Comment:
매개변수로 받아오는 viewHolder 타입을 이런식으로 바꾸면 as를 쓸필요가 없어요!
```suggestion
    private fun getView(viewHolder: ContentDetailDiaryAdapter.ContentDetailDiaryViewHolder): View {
        return viewHolder.binding.clItemDiaryTop
    }
```